service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/AuthenticationServiceShiroImpl.java

```diff
@@ -132,6 +133,13 @@ public class AuthenticationServiceShiroImpl implements AuthenticationService {
 
     private final UserService userService;
 
+    @Inject
+    private Set<LoginCredentialsHandler> loginCredentialsHandlers;
+
+    @Inject
+    private Set<SessionCredentialsHandler> sessionCredentialsHandlers;
+
```

Comment:
To distinguish the two types of authentication that we have.

The first login (handled by `login` with `LoginCredentials` and meant to create a session on EC side) and the continuation of that session (handled by `authenticate` with `SessionCredentials` to recover that session). 

So following the same logic of `AuthenticationCredentials` splitting into `LoginCredentials` and `SessionCredentials`.

This API design is something that we can change, if we want but maybe it would be better change in another PR to refactor the API and then apply the changes on this PR.

Reply:
I'm not 100% sure regarding this mechanism, and would probably review it. Still, for the purpose of this PR I'd propose at least to split the method CredentialsHandler.canProcess into:

* canLogin
* canAuthenticate

and handle the difference there - you can then invoke each method from the appropriate AuthenticationService branch, without creating too many classes